api/src/transactions.rs

```diff
@@ -1380,10 +1380,11 @@ impl TransactionsApi {
         let version = ledger_info.version();
 
         // Ensure that all known statuses return their values in the output (even if they aren't supposed to)
-        let exe_status = match output.status().clone() {
-            TransactionStatus::Keep(exec_status) => exec_status,
-            TransactionStatus::Discard(status) => ExecutionStatus::MiscellaneousError(Some(status)),
-            _ => ExecutionStatus::MiscellaneousError(None),
+        let exe_status = match vm_status.clone().keep_or_discard() {
+            Ok(kept_vm_status) => kept_vm_status.into(),
+            Err(discarded_vm_status) => {
+                ExecutionStatus::MiscellaneousError(Some(discarded_vm_status))
+            },
```

Comment:
is this intentioinal?

Reply:
yes. if we turn on feature flag, we need to read from vm_status to get the error details